packages/cdk-route-splitting/lib/index.ts

```diff
@@ -0,0 +1,153 @@
+import * as cdk from '@aws-cdk/core';
+import * as cloudfront from '@aws-cdk/aws-cloudfront';
+import { HttpOrigin } from '@aws-cdk/aws-cloudfront-origins';
+import * as dynamodb from '@aws-cdk/aws-dynamodb';
+import * as apigateway from '@aws-cdk/aws-apigateway';
+import * as route53 from '@aws-cdk/aws-route53';
+import * as acm from '@aws-cdk/aws-certificatemanager';
+import * as iam from '@aws-cdk/aws-iam';
+import * as route53targets from '@aws-cdk/aws-route53-targets';
+
+const lambda = require('@aws-cdk/aws-lambda');
+
+export interface CdkRouteSplittingProps {
+  hostedZoneName: string;
+  hostedZoneId: string;
+  projectName: string;
+  originSourceDomain: string;
+  partitionKey: string;
+  stage: string;
+  originNotFoundUrl: string;
+}
+
+export class CdkRouteSplitting extends cdk.Construct {
+
+  constructor(scope: cdk.Construct, id: string, props: CdkRouteSplittingProps) {
+    super(scope, id);
+
+    const lambdaRole = new iam.Role(this, `${props.stage}-${props.projectName}-edge-function-role`, {
+      assumedBy: new iam.CompositePrincipal(new iam.ServicePrincipal('lambda.amazonaws.com'), new iam.ServicePrincipal('edgelambda.amazonaws.com')),
+    });
+    lambdaRole.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName("service-role/AWSLambdaBasicExecutionRole"));
+    lambdaRole.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName("service-role/AWSLambdaVPCAccessExecutionRole")); 
+
+    const hostedZone = route53.HostedZone.fromHostedZoneAttributes(this, "HostedZone", {
+      zoneName: props.hostedZoneName,
+      hostedZoneId: props.hostedZoneId,
+    });
+
+    let domainName: string;
+    if(props.stage == "prod") {
+      domainName = `${props.projectName}.${props.hostedZoneName}`;
+    } else {
+      domainName = `${props.stage}-${props.projectName}.${props.hostedZoneName}`;
+    }
+
+    const cert = new acm.Certificate(this, "Certificate", {
```

Comment:
Generally since certs can take so long to generate, it is better to have them in a separate stack and import them.